src/solver/toolkit.c

```diff
@@ -1920,31 +1920,36 @@ EXPORT_TOOLKIT int swmm_getNodePollut(int index, SM_NodePollut type, double **po
                 *pollutArray = result;
                 *length = Nobjects[POLLUT];
             } break;
-	    case SM_NODECIN:
-	    {
-		for (p=0; p < Nobjects[POLLUT]; p++)
-		{
-		    result[p] = Node[index].inQual[p];
-		} *pollutArray = result;
-	    } break;
-	    case SM_NODEREACTORC:
-	    {
-		for (p=0; p < Nobjects[POLLUT]; p++)
-		{
-		    result[p] = Node[index].reactorQual[p];
-		} *pollutArray = result;
-	    } break;
+            case SM_NODECIN:
+            {
+                for (p=0; p < Nobjects[POLLUT]; p++)
+                {
+                    result[p] = Node[index].inQual[p];
+                }
+                *pollutArray = result;
+                *length = Nobjects[POLLUT];
```

Comment:
Pulling length assignment out of the switch statement would make the code more DRY since we wouldn't need the same line in each switch case.